### PR TITLE
 SIL: Assert that SILFunctionTypes are only @pseudogeneric when they have generic arguments.

### PR DESCRIPTION
**Pull request by jckarter**

### CODE REVIEW

**Comment by jckarter** [comment]
@swift-ci Please test

File: test/SILGen/objc_block_to_func_to_block.swift [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -import-objc-header %S/Inputs/objc_block_to_func_to_block.h -emit-silgen -verify %s
```
Comment (jrose-apple) [inline test/SILGen/objc_block_to_func_to_block.swift]: This is going to use the real overlays with the mock SDK, which isn't really safe.

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@swift-ci Please test

**Pull request merged**